cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -72,42 +74,59 @@ bool AzureOptions::Equals(const AzureOptions& other) const {
   return false;
 }
 
-void AzureOptions::SetUrlsForAccountName(const std::string& account_name) {
-  if (this->backend == AzureBackend::kAzurite) {
-    account_blob_url_ = "http://127.0.0.1:10000/" + account_name + "/";
-    account_dfs_url_ = "http://127.0.0.1:10000/" + account_name + "/";
-  } else {
-    account_dfs_url_ = "https://" + account_name + ".dfs.core.windows.net/";
-    account_blob_url_ = "https://" + account_name + ".blob.core.windows.net/";
+namespace {
+std::string BuildBaseUrl(const std::string& scheme, const std::string& authority,
+                         const std::string& account_name) {
+  std::string url;
+  url += scheme + "://";
+  if (!authority.empty()) {
```

Comment:
I think that we should reject empty authority because it builds no host part URL.

Reply:
My reasoning here is that there are many validations I could apply, but the most complete validation will come from the error messages triggered by using the filesystem with a bogus URL with 3 slashes after `https:`.